includes/class-llms-form-templates.php

```diff
@@ -0,0 +1,680 @@
+					'required'         => false,
+					'label_show_empty' => true,
+				)
+			),
+			'width'   => 33.33,
+		);
+
+		return $this->wrap_columns( $cols );
+
+	}
+
+	/**
+	 * Retrieve block(s) for the email address row.
+	 *
+	 * @since [version]
+	 *
+	 * @param string $location Form location. Accepts template options passed to `$this->get_template()`.
+	 * @return array
+	 */
+	protected function get_row_email( $location ) {
+
+		$cols = array();
+
+		$cols[] = $this->get_block(
+			'user-email',
+			array(
+				'field' => 'email',
+				'id'    => 'email_address',
+				'label' => __( 'Email Address', 'lifterlms' ),
+				'match' => 'email_address_confirm',
+			)
+		);
+
+		$option = get_option( sprintf( 'lifterlms_user_info_field_email_confirmation_%s_visibility', $location ) );
+		if ( ! llms_parse_bool( $option ) ) {
+			return $cols;
+		}
+
+		$cols[] = $this->get_block(
+			'user-email-confirm',
+			array(
+				'field'          => 'email',
+				'id'             => 'email_address_confirm',
+				'label'          => __( 'Confirm Email Address', 'lifterlms' ),
+				'match'          => 'email_address',
+				'data_store_key' => false,
+			)
+		);
+
+		return $this->wrap_columns( $cols, 'logged_out' );
+
+	}
+
+	/**
+	 * Retrieve blocks for the name row.
+	 *
+	 * Returns an empty array if names are disabled by legacy settings.
+	 *
+	 * @since [version]
+	 *
+	 * @param string $location Form location. Accepts template options passed to `$this->get_template()`.
+	 * @return array
+	 */
+	protected function get_row_names( $location ) {
+
+		$cols = array();
+
+		$option = get_option( sprintf( 'lifterlms_user_info_field_names_%s_visibility', $location ), 'required' );
+		if ( 'hidden' === $option ) {
+			return $cols;
+		}
+
+		$cols[] = $this->get_block(
+			'user-first-name',
+			array(
+				'id'       => 'first_name',
+				'label'    => __( 'First Name', 'lifterlms' ),
+				'required' => ( 'required' === $option ),
+			)
+		);
+
+		$cols[] = $this->get_block(
+			'user-last-name',
+			array(
+				'id'       => 'last_name',
+				'label'    => __( 'Last Name', 'lifterlms' ),
+				'required' => ( 'required' === $option ),
+			)
+		);
+
+		return $this->wrap_columns( $cols );
+
+	}
+
+	/**
+	 * Retrieve blocks for the password row.
+	 *
+	 * @since [version]
+	 *
+	 * @return array
+	 */
+	protected function get_row_password() {
+
+		$cols = array();
+
+		$cols[] = $this->get_block(
+			'user-password',
+			array(
+				'field' => 'password',
+				'id'    => 'password',
+				'label' => __( 'Password', 'lifterlms' ),
+				'match' => 'password_confirm',
+			)
+		);
+
+		$cols[] = $this->get_block(
+			'user-password-confirm',
+			array(
+				'field'          => 'password',
+				'id'             => 'password_confirm',
+				'label'          => __( 'Confirm Password', 'lifterlms' ),
+				'match'          => 'password',
+				'data_store_key' => false,
+			)
+		);
+
+		return $this->wrap_columns( $cols, 'logged_out' );
```

Comment:
These fields are used in the dashboard edit account form as well, so they should be visible to all... right?